torch/utils/dlpack.py

```diff
@@ -25,3 +15,38 @@
 The dlpack shares the tensors memory.
 Note that each dlpack can only be consumed once.
 """)
+
+
+def from_dlpack(ext_tensor: Any) -> torch.Tensor:
+    """from_dlpack(ext_tensor) -> Tensor
+
+    Convers a tensor from a external library into a ``torch.Tensor``
+    by means of the ``__dlpack__`` protocol.
+
+    The tensor will share the memory with the object represented
+    in the dlpack.
+
+    In order to keep backward compatibility, this function also admits
```

Comment:
Backward compatibility with what?

Reply:
old's pytorch `from_dlpack` implementation, previously it took a dlpack capsule object, now we pass a, object with `__dlpack__` protocol, so we accept both objects in order to not break existing code.

This can be changed.

Reply:
Accepting both seems great. I just don't think most people will understand the motivation being driven by BC, so we probably shouldn't mention it.